src/modules/cpu/kernel/spectrogram.hpp

```diff
@@ -0,0 +1,248 @@
+    {
+        Rpp32f phase = a * (t + 0.5);
+        output[t] = (0.5 * (1.0 - std::cos(phase)));
+    }
+}
+
+inline Rpp32s get_num_windows(Rpp32s length, Rpp32s windowLength, Rpp32s windowStep, bool centerWindows)
+{
+    if (!centerWindows)
+        length -= windowLength;
+    return ((length / windowStep) + 1);
+}
+
+inline Rpp32s get_idx_reflect(Rpp32s idx, Rpp32s lo, Rpp32s hi)
+{
+    if (hi - lo < 2)
+        return hi - 1;
+    for (;;)
+    {
+        if (idx < lo)
+            idx = 2 * lo - idx;
+        else if (idx >= hi)
+            idx = 2 * hi - 2 - idx;
+        else
+            break;
+    }
+    return idx;
+}
+
+RppStatus spectrogram_host_tensor(Rpp32f *srcPtr,
+                                  RpptDescPtr srcDescPtr,
+                                  Rpp32f *dstPtr,
+                                  RpptDescPtr dstDescPtr,
+                                  Rpp32s *srcLengthTensor,
+                                  bool centerWindows,
+                                  bool reflectPadding,
+                                  Rpp32f *windowFunction,
+                                  Rpp32s nfft,
+                                  Rpp32s power,
+                                  Rpp32s windowLength,
+                                  Rpp32s windowStep,
+                                  RpptSpectrogramLayout layout)
+{
+    Rpp32s windowCenterOffset = 0;
+    bool vertical = (layout == RpptSpectrogramLayout::FT);
+    if (centerWindows)
+        windowCenterOffset = windowLength / 2;
+    if (nfft == 0.0f)
+        nfft = windowLength;
+    Rpp32s numBins = nfft / 2 + 1;
+    const Rpp32f mulFactor = TWO_PI / nfft;
+    Rpp32u hStride = dstDescPtr->strides.hStride;
+    Rpp32s alignedNfftLength = nfft & ~7;
+    Rpp32s alignedNbinsLength = numBins & ~7;
+    Rpp32s alignedWindowLength = windowLength & ~7;
+    Rpp32f cosf[numBins * nfft];
+    Rpp32f sinf[numBins * nfft];
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(8)
+    for (Rpp32s k = 0; k < numBins; k++)
+    {
+        Rpp32f* cosfTemp = cosf + (k * nfft);
+        Rpp32f* sinfTemp = sinf + (k * nfft);
+        __m256 pJN = _mm256_set_ps(7,6,5,4,3,2,1,0);
+        __m256 pMulFactorN = _mm256_set1_ps(k*mulFactor);
+        __m256 pAddFactorN = _mm256_set1_ps(8);
+        Rpp32s i = 0;
+        for (; i < alignedNfftLength; i += 8) {
+            __m256 pSrc = _mm256_mul_ps(pJN,pMulFactorN);
+            __m256 pSin, pCos;
+            sincos_ps(pSrc, &pSin, &pCos);
+            pSin = _mm256_mul_ps(avx_pm1, pSin);
+            pJN = _mm256_add_ps(pJN,pAddFactorN);
+            _mm256_storeu_ps(cosfTemp, pCos);
+            _mm256_storeu_ps(sinfTemp, pSin);
+            cosfTemp += 8;
+            sinfTemp += 8;
+        }
+        for (; i < nfft; i++) {
+            *cosfTemp++ = std::cos(k * i * mulFactor);
+            *sinfTemp++ = -std::sin(k * i * mulFactor);
+        }
+    }
+    std::vector<Rpp32f> windowFn;
```

Comment:
We can add empty lines between 2 logical code blocks similar to here
We are computing sin and cos values from L91 to L102 which is one logical code block and codes from L103 is another code block, So we can add empty line here

Please add empty lines to all such cases in this file